setup.py

```diff
@@ -1,92 +1,28 @@
 #!/usr/bin/env python
 # -*- encoding: utf-8 -*-
-from __future__ import absolute_import
-
-import io
-import re
-from glob import glob
-from os import path, environ
-
-from setuptools import find_packages
+import os
 
 # Include extensions only when not on readthedocs.org
-if environ.get('READTHEDOCS', None) == 'True':
+if os.environ.get('READTHEDOCS', None) == 'True':
     from setuptools import setup
     from distutils.core import Extension
     extensions = []
 else:
     from numpy.distutils.core import setup, Extension
+
+    # add MinGW to Windows only, if needed
+    if os.name == 'nt':
+        sfn = os.path.join(os.path.dirname(__file__), 'setup.cfg')
+        stxt = open(sfn).read()
+        if '[build_ext]' not in stxt:
+            with open(sfn, 'a') as f:
+                f.write("\n[build_ext]\ncompiler = mingw32")
```

Comment:
This seems to hardcode in the mingw32 compiler. What about mingw64, etc? There's got to be a better way to detect the appropriate compiler on windows? Maybe not.

Reply:
This is a workaround for Numpy distutils behavior with MinGW on Windows, it is a longstanding issue (many years). Numpy 1.16 introduced new functionality I am using in other projects. I didn't want to introduce a Numpy &ge; 1.16 requirement at this time.

Reply:
That is, with Numpy &ge; 1.16 this setup.py blurb is not needed. My Fortran + Python projects have a one-liner setup.py just like plain Python projects:
```python
import setuptools; setuptools.setup()
```
and everything else is in setup.cfg and other static files (mypy.ini, .flake8, .coveragerc, etc.)

Reply:
`mingw32` actually is also for MinGW64, which is what I normally use for Gfortran on native Windows

Reply:
Thanks for the info! This will help when I dig in with my windows machine.